src/lib.rs

```diff
@@ -76,5 +76,11 @@ pub use tls::TlsStream;
 pub use crate::consumer::{Consumer, ConsumerBuilder};
 pub use crate::error::Error;
 pub use crate::producer::Producer;
-pub use crate::proto::Reconnect;
-pub use crate::proto::{Job, JobBuilder};
+pub use crate::proto::{Client, Job, JobBuilder, Reconnect};
+
+#[cfg(feature = "ent")]
+#[cfg_attr(docsrs, doc(cfg(feature = "ent")))]
+pub use crate::proto::{
+    Batch, BatchBuilder, BatchHandle, BatchStatus, CallbackState, JobState, Progress,
+    ProgressUpdate, ProgressUpdateBuilder,
+};
```

Comment:
The root of the crate is starting to have a _lot_ of types in it. How would you feel about making this:
```suggestion
/// Constructs only available with the enterprise version of Faktory. 
pub mod ent {
    pub use crate::proto::{
        Batch, BatchBuilder, BatchHandle, BatchStatus, CallbackState, JobState, Progress,
        ProgressUpdate, ProgressUpdateBuilder,
    };
}
```

Reply:
Or, alternatively, not exposing most of these at the root at all?

Reply:
I really like this namespacing!

Reply:
Indeed, much better, less noise
![image](https://github.com/jonhoo/faktory-rs/assets/117771945/a64fb56d-5ba2-4ee4-a03e-cf3b0ec215a1)